tests/functional/c/consider/consider_using_augmented_assign.py

```diff
@@ -0,0 +1,62 @@
+"""Tests for consider-using-augmented-assign."""
+
+# pylint: disable=invalid-name,too-few-public-methods,import-error,consider-using-f-string
+
+from unknown import Unknown
+
+x = 1
+x = x + 1  # [consider-using-augmented-assign]
+x = 1 + x  # [consider-using-augmented-assign]
+x, y = 1 + x, 2 + x
+# We don't warn on intricate expressions as we lack knowledge
+# of simplifying such expressions which is necessary to see
+# if they can become augmented
+x = 1 + x - 2
+x = 1 + x + 2
+
+# For anything other than a float or an int we only want to warn on
+# assignments where the 'itself' is on the left side of the assignment
+my_list = [2, 3, 4]
+my_list = [1] + my_list
+
+
```

Comment:
Do we really need this? We don't do any operator checking in the utility? The `in` and `not in` are `nodes.Comparse` so they don't feature in this utility.

Reply:
Well we did have ``>>=`` suggested in pygame, imo it does not hurt to check.

Reply:
> `>>=`

That's valid right?

```consol
Python 3.10.0 (default, Oct 27 2021, 16:20:25) [Clang 13.0.0 (clang-1300.0.29.3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 2 << 6
128
>>> x = 2
>>> x <<= 6
>>> x
128
```

Reply:
I added the tests that make sense to me. Checking for `is not` uses a different node so I didn't include variants of those. I did do `x > 3` as it is visually more like "algebra" or `BinOps`.

Reply:
> That's valid right?

Sure ! I did not see the almost identical tests from line 64 below where you tested most of the thing I suggested. 

So the remaining one would be those:
```
x = Unknown
x = x not in 3
x = x not in [3]
x = x in 3
x = x in [3]
x = x is 3
x = x is not 3
x = x is 3
x = x is not 3
x = x and 3
x = x or 3
```
Kinda agree this might not be necessary to tests them.

Reply:
Yeah I mean that's just assignments of booleans. For those we have our whole test suite as a regressions 😄
We wouldn't be able to pass any run if we started flagging those 😅